src/components/NewMovie/NewMovie.js

```diff
@@ -1,13 +1,214 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 
 export class NewMovie extends Component {
-  state = {};
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    checkTitle: false,
+    checkImgUrl: false,
+    checkImdbUrl: false,
+    checkImdbId: false,
+    isButtonDisabled: true,
+  };
+
+  setTitle = (e) => {
+    this.setState({
+      title: e.target.value,
+      checkTitle: false,
+    });
+  };
+
+  setDescription = (e) => {
+    this.setState({
+      description: e.target.value,
+    });
+  };
+
+  setImgUrl = (e) => {
+    this.setState({
+      imgUrl: e.target.value,
+    });
+  };
+
+  setImdbUrl = (e) => {
+    this.setState({
+      imdbUrl: e.target.value,
+
+    });
+  };
+
+  setImdbId = (e) => {
+    this.setState({
+      imdbId: e.target.value,
+    });
+  };
+
+  handleActiveButton = () => {
+    const {
+      title, description, imgUrl, imdbUrl, imdbId,
+    } = this.state;
+
+    if (title.trim().length > 0
+      && description.trim().length > 0
+      && imgUrl.trim().length > 0
+      && imdbUrl.trim().length > 0
+      && imdbId.trim().length > 0
+    ) {
+      this.setState({ isButtonDisabled: false });
+    }
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    this.setState((state) => {
+      const {
+        title, description, imgUrl, imdbUrl, imdbId,
+      } = state;
+      const { addMovie, validation } = this.props;
+
+      if (title.length === 0 || title.trim() === '') {
+        return { checkTitle: true };
+      }
+
+      if (!validation.test(imgUrl)) {
+        return { checkImgUrl: true };
+      }
+
+      if (!validation.test(imdbUrl)) {
+        return { checkImdbUrl: true };
+      }
+
+      if (imdbId.length === 0 || imdbId.trim() === '') {
+        return { checkImdbId: true };
+      }
+
+      addMovie({
+        title,
+        description,
+        imgUrl,
+        imdbUrl,
+        imdbId,
+      });
+
+      return {
+        title: '',
+        description: '',
+        imgUrl: '',
+        imdbUrl: '',
+        imdbId: '',
+        checkTitle: false,
+        checkImgUrl: false,
+        checkImdbUrl: false,
+        checkImdbId: false,
+      };
+    });
+  }
+
+  onBlur = (event) => {
+    event.target.style.borderColor = 'red';
+  };
 
   render() {
+    const { title, description, imgUrl, imdbUrl, imdbId, checkTitle,
+      checkImgUrl, checkImdbUrl, checkImdbId, isButtonDisabled } = this.state;
```

Comment:
```suggestion
    const { 
      title, description, imgUrl, imdbUrl, imdbId, checkTitle,
      checkImgUrl, checkImdbUrl, checkImdbId, isButtonDisabled 
    } = this.state;
```
move each one on the new line

Reply:
👌🏻